packages/block-library/src/media-text/edit.js

```diff
@@ -145,9 +150,42 @@ function MediaTextEdit( { attributes, isSelected, setAttributes } ) {
 		rel,
 		verticalAlignment,
 		allowedBlocks,
+		useFeaturedImage,
 	} = attributes;
 	const mediaSizeSlug = attributes.mediaSizeSlug || DEFAULT_MEDIA_SIZE_SLUG;
 
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const featuredImageMedia = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+
+	const featuredImageURL = useFeaturedImage
+		? featuredImageMedia?.source_url
+		: '';
+	const featuredImageAlt = useFeaturedImage
+		? featuredImageMedia?.alt_text
+		: '';
+
+	const toggleUseFeaturedImage = () => {
+		setAttributes( {
+			imageFill: false,
+			mediaType: 'image',
```

Comment:
By setting `mediaType` to `undefined`, we will be able to remove the `img` tag that does not have attributes from the saved markup. Instead, I believe that it's possible to inject an `img` tag in the dynamic rendering when `useFeaturedImage` is `true`.

We should be able to use `preg_replace` to inject the `img` element as a string right after the opening tag of the `figure` element. This approach is also used in [the Image block](https://github.com/WordPress/gutenberg/blob/b8474c83aa37c7ab299b4f251a0f96ad598c5f95/packages/block-library/src/image/index.php#L230). In the future, if the HTML API becomes able to inject arbitrary elements, it will be possible to refactor it at that time.

Reply:
Thank you.
I'm trying, but I can't get it to work, the image tag is only rended on the front. I must be missing a step, or there is something that we did not consider.
I will push the changes though so they can be tested.

Reply:
I mean, is is it just because the <figure> is not identical in the editor and front?